web/src/views/InvestigationDetail.vue

```diff
@@ -222,6 +236,32 @@ export default defineComponent({
       pinFeature = null;
     }
 
+    function movePinNoteCards() {
+      if (!featureLayer || !pinFeature || !map.value) { return; }
+      pinFeature.data().forEach((pinPoint: any) => {
+        const { x, y } = pinPoint;
+        const newScreenCoords = pinFeature.featureGcsToDisplay({ x, y });
+        const note = pinNotes.value.find((pinNote) => pinNote.id === pinPoint.id);
+        const {
+          left, top, width, height,
+        } = map.value?.getBoundingClientRect() || {
+          left: 0, top: 0, width: 0, height: 0,
+        };
+        note.notePositionX = newScreenCoords.x + left;
+        note.notePositionY = newScreenCoords.y + top;
+        if (note.notePositionX > left + width
+            || note.notePositionY > top + height
+            || note.notePositionX < left
+            || note.notePositionY < top) {
+          note.inBounds = false;
+        } else {
+          note.inBounds = true;
+        }
+      });
+    }
+
+    const debounceMovePinNoteCards = debounce(movePinNoteCards, 50);
```

Comment:
I'm curious about why this is needed. I just read up on the concept. [Here's a great blog post](https://stoilsky.com/js-debounce-function/). It's probably common knowledge in the JavaScript world, but I learned something new. Is the callback function too heavy to do without this debounce?

Reply:
It's probably not necessary. I get nervous when doing DOM manipulation, as my understanding is that its slow, although maybe I'm wrong, or being overly cautious here.

Not debouncing doesn't seem to have an impact on performance at the moment, and the UX gain is very noticeable, so I'll take that out.

Reply:
See 4d9a36a